extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/entitymanager/TransactionScopedEntityManager.java

```diff
@@ -112,20 +113,29 @@ private boolean isInTransaction() {
     @Override
     public void persist(Object entity) {
         try (EntityManagerResult emr = getEntityManager()) {
+            if (!emr.allowModification) {
+                throw new TransactionRequiredException("Transaction is not active");
```

Comment:
How about `Transaction is not active: add @Transactional to your method`?

Reply:
This would not work at the moment, right?

Reply:
@mkouba why not?

Reply:
Well, it will work once https://github.com/quarkusio/quarkus/pull/2629 is merged. Tests are not CDI beans ATM.